### PR TITLE
Upgrade google api client

### PR DESCRIPTION
**Pull request by blowmage**
WIP of the Google API Client upgrade.

### CODE REVIEW

**Comment by quartzmo** [comment]
I'm going to start work on Translate.

File: lib/gcloud/resource_manager/service.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+      def service
+        return mocked_service if mocked_service
+        @service
+      end
+      attr_accessor :mocked_service
+
+      ##
+      # Returns API::ListProjectsResponse
+      def list_project filter: nil, token: nil, max: nil
```
Comment (quartzmo) [inline lib/gcloud/resource_manager/service.rb]: Should this be `list_projects` instead?

Comment (blowmage) [inline lib/gcloud/resource_manager/service.rb]: It could be. I wasn't trying to make improvements in that commit, just convert the client.

Comment (quartzmo) [inline lib/gcloud/resource_manager/service.rb]: Gotcha!

**Comment by quartzmo** [comment]
Can `lib/gcloud/resource_manager/errors.rb` be deleted?

**Comment by quartzmo** [comment]
I think `ApiError` will still be needed, although it may not have HTTP attributes such as `status` In Vision I have encountered an error object nested in the response object, like this: 

```sh
#<Google::Apis::VisionV1::AnnotateImageResponse:0x007fa6b65a8048 @error=#<Google::Apis::VisionV1::Status:0x007fa6b6619d10 @code=3, @message="image-annotator::Malformed request.: Image processing error!">>
```

**Comment by blowmage** [comment]
We will have additional work to handle errors appropriately in the 0.9 client. That particular response is just how vision API works. https://cloud.google.com/vision/reference/rest/v1/images/annotate#annotateimageresponse

**Comment by quartzmo** [comment]
I have signed the CLA.

**Comment by quartzmo** [comment]
This is a big PR and although all tests are passing, we need to follow up with extensive manual testing. Should we do that before merging?

**Comment by blowmage** [comment]
I agree. I think there is an argument that can be made to merge this while we test so that the acceptance tests are running on CI.

**Comment by quartzmo** [comment]
That's a great point. Merging now.

**Comment by blowmage** [comment]
Here are the issues this PR closes: #201, #269, #490, #734, #735

**Pull request merged**